google-ads-examples/src/main/java/com/google/ads/googleads/examples/remarketing/AddMerchantCenterDynamicRemarketingCampaign.java

```diff
@@ -0,0 +1,411 @@
+import com.google.ads.googleads.v2.resources.Campaign.ShoppingSetting;
+import com.google.ads.googleads.v2.services.AdGroupAdOperation;
+import com.google.ads.googleads.v2.services.AdGroupAdServiceClient;
+import com.google.ads.googleads.v2.services.AdGroupCriterionOperation;
+import com.google.ads.googleads.v2.services.AdGroupCriterionServiceClient;
+import com.google.ads.googleads.v2.services.AdGroupOperation;
+import com.google.ads.googleads.v2.services.AdGroupServiceClient;
+import com.google.ads.googleads.v2.services.AssetOperation;
+import com.google.ads.googleads.v2.services.AssetServiceClient;
+import com.google.ads.googleads.v2.services.CampaignOperation;
+import com.google.ads.googleads.v2.services.CampaignServiceClient;
+import com.google.ads.googleads.v2.services.MutateAdGroupAdsResponse;
+import com.google.ads.googleads.v2.services.MutateAdGroupCriteriaResponse;
+import com.google.ads.googleads.v2.services.MutateAdGroupsResponse;
+import com.google.ads.googleads.v2.services.MutateAssetsResponse;
+import com.google.ads.googleads.v2.services.MutateCampaignsResponse;
+import com.google.ads.googleads.v2.utils.ResourceNames;
+import com.google.common.collect.ImmutableList;
+import com.google.common.io.ByteStreams;
+import com.google.protobuf.BoolValue;
+import com.google.protobuf.ByteString;
+import com.google.protobuf.BytesValue;
+import com.google.protobuf.Int32Value;
+import com.google.protobuf.Int64Value;
+import com.google.protobuf.StringValue;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.net.URL;
+
+/**
+ *  Creates a shopping campaign associated with an existing merchant center account, along with a
+ *  related ad group and dynamic display ad, and targets a user list for remarketing purposes.
+ */
+public class AddMerchantCenterDynamicRemarketingCampaign {
+  private static class AddMerchantCenterDynamicRemarketingCampaignParams extends CodeSampleParams {
+
+    @Parameter(names = ArgumentNames.CUSTOMER_ID, required = true)
+    private long customerId;
+
+    @Parameter(names = ArgumentNames.MERCHANT_CENTER_ACCOUNT_ID, required = true)
+    private long merchantCenterAccountId;
+
+    @Parameter(names = ArgumentNames.CAMPAIGN_BUDGET_ID, required = true)
+    private long campaignBudgetId;
+
+    @Parameter(names = ArgumentNames.USER_LIST_ID, required = true)
+    private long userListId;
+  }
+
+  public static void main(String[] args) throws IOException {
+    AddMerchantCenterDynamicRemarketingCampaignParams params =
+      new AddMerchantCenterDynamicRemarketingCampaignParams();
+    if (!params.parseArguments(args)) {
+
+      // Either pass the required parameters for this example on the command line, or insert them
+      // into the code here. See the parameter class definition above for descriptions.
+      params.customerId = Long.parseLong("INSERT_CUSTOMER_ID_HERE");
+      params.merchantCenterAccountId = Long.parseLong("INSERT_MERCHANT_CENTER_ID_HERE");
```

Comment:
minor issue
      params.merchantCenterAccountId = Long.parseLong("**INSERT_MERCHANT_CENTER_ACCOUNT_ID_HERE**");

as consistent with "AddShoppingSmartAd" and "AddShoppingProductAd" sample.